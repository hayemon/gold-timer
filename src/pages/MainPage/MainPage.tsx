import { Card, Col, List, Row } from 'antd'
import { Board } from 'components'
import useTimer from '../../hooks/useTimer'

const MainPage: React.FC = () => {
  const { inactive, lastValue, rewards, totalOverdrives } = useTimer()

  return (
    <div className='container'>
      <Row>
        <Col span={6}>
          <Card title='Stats'>
            <List>
              {inactive && <List.Item>Inactive</List.Item>}
              <List.Item>Previous Max {lastValue}</List.Item>
              <List.Item>Overldrives Count {totalOverdrives}</List.Item>
            </List>
          </Card>
        </Col>
        <Col span={12}>
          <Board />
        </Col>
        <Col span={6}>
          <Card title='Rewards'>
            <List>
              {rewards.map((x, i) => (
                <List.Item key={i}>{x}</List.Item>
              ))}
            </List>
          </Card>
        </Col>
      </Row>
    </div>
  )
}

export default MainPage
