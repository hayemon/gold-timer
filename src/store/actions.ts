export { addReward, incrementTotalOverdrives } from './slices/statSlice'
export {
  setInactive,
  setLastValue,
  setOverdriveCount,
  setPressCount,
  setTimerCount,
} from './slices/timerSlice'
