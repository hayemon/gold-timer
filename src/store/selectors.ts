export { selectStat } from './slices/statSlice'
export { selectTimer } from './slices/timerSlice'
