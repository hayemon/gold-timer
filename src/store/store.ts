import {
  combineReducers,
  configureStore,
  PreloadedState,
} from '@reduxjs/toolkit'
import statReducer from './slices/statSlice'
import timerReducer from './slices/timerSlice'

const rootReducer = combineReducers({
  timer: timerReducer,
  stat: statReducer,
})

export const createStore = (preloadedState?: PreloadedState<RootState>) =>
  configureStore({
    reducer: rootReducer,
    preloadedState,
  })

export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof createStore>
export type AppDispatch = AppStore['dispatch']
