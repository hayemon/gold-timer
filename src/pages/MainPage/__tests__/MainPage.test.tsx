import { screen } from '@testing-library/react'
import { addReward } from 'store/actions'
import { renderWithProviders } from 'store/testUtils'
import MainPage from '../MainPage'

describe('MainPage', () => {
  it('renders', () => {
    const { asFragment } = renderWithProviders(<MainPage />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('add rewards', () => {
    const { store } = renderWithProviders(<MainPage />)

    store.dispatch(addReward('Test'))
    expect(store.getState().stat.rewards[0]).toBe('Test')
    screen.getByText('Test')

    store.dispatch(addReward('Test'))
    expect(store.getState().stat.rewards.toString()).toBe('Test,Test')
  })
})
