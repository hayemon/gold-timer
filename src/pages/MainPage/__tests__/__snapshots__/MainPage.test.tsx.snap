// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-6"
      >
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Stats
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-list ant-list-split"
            >
              <div
                class="ant-spin-nested-loading"
              >
                <div
                  class="ant-spin-container"
                >
                  <li
                    class="ant-list-item ant-list-item-no-flex"
                  >
                    Previous Max 0
                  </li>
                  <li
                    class="ant-list-item ant-list-item-no-flex"
                  >
                    Overldrives Count 0
                  </li>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-12"
      >
        <div
          class="board-container"
        >
          <div
            class="treasure-box-container"
          >
            <h1
              class="ant-typography"
            >
              0 Coins!!!
            </h1>
            <button
              class="ant-btn"
              style="width: 200px; height: 200px; margin: 0px; padding: 0px;"
              type="button"
            >
              <div
                class="treasure-box"
              />
            </button>
          </div>
          <div
            class="indicator-container"
          />
        </div>
      </div>
      <div
        class="ant-col ant-col-6"
      >
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Rewards
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-list ant-list-split"
            >
              <div
                class="ant-spin-nested-loading"
              >
                <div
                  class="ant-spin-container"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
